pennylane/templates/tensornetworks/mera.py

```diff
@@ -0,0 +1,258 @@
+    wires_list = []
+    wires_list.append(list(wires[0:n_block_wires]))
+    highest_index = n_block_wires
+    for i in range(n_layers - 1):
+        n_elements_pre = 2 ** ((i + 1) // 2)
+        if i % 2 == 0:
+            new_list = []
+            list_len = len(wires_list)
+            for j in range(list_len - n_elements_pre, list_len):
+                new_wires = [
+                    wires[i] for i in range(highest_index, highest_index + n_block_wires // 2)
+                ]
+                highest_index += n_block_wires // 2
+                new_list.append(wires_list[j][0 : n_block_wires // 2] + new_wires)
+                new_wires = [
+                    wires[i] for i in range(highest_index, highest_index + n_block_wires // 2)
+                ]
+                highest_index += n_block_wires // 2
+                new_list.append(new_wires + wires_list[j][n_block_wires // 2 : :])
+            wires_list = wires_list + new_list
+        else:
+            list_len = len(wires_list)
+            new_list = []
+            for j in range(list_len - n_elements_pre, list_len - 1):
+                new_list.append(
+                    wires_list[j][n_block_wires // 2 : :]
+                    + wires_list[j + 1][0 : n_block_wires // 2]
+                )
+            new_list.append(
+                wires_list[j + 1][n_block_wires // 2 : :]
+                + wires_list[list_len - n_elements_pre][0 : n_block_wires // 2]
+            )
+            wires_list = wires_list + new_list
+    return wires_list[::-1]
+
+
+class MERA(Operation):
+    """The MERA template broadcasts an input circuit across many wires following the
+    architecture of a multi-scale entanglement renormalization ansatz tensor network.
+    This architecture can be found in `arXiv:quant-ph/0610099 <https://arxiv.org/abs/quant-ph/0610099>`
+    and closely resembles `quantum convolutional neural networks <https://arxiv.org/abs/1810.03787>`_.
+
+    The argument ``block`` is a user-defined quantum circuit. Each ``block`` may depend on a different set of parameters.
+    These are passed as a list by the ``template_weights`` argument.
+
+    For more details, see *Usage Details* below.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        n_block_wires (int): number of wires per block
+        block (Callable): quantum circuit that defines a block
+        n_params_block (int): the number of parameters in a block
+        template_weights (Sequence): list containing the weights for all blocks
+
+    .. UsageDetails::
+
+        In general, the block takes D parameters and **must** have the following signature:
+
+        .. code-block:: python
+
+            unitary(parameter1, parameter2, ... parameterD, wires)
+
+        For a block with multiple parameters, ``n_params_block`` is equal to the number of parameters in ``block``.
+        For a block with a single parameter, ``n_params_block`` is equal to the length of the parameter.
```

Comment:
What does "length of the parameter" mean? That this can generally be an array?

Reply:
Yup, this can generally be an array. By length I just mean len(parameter), or the first dimension of the array.